app/workers/fasp/account_search_worker.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Fasp::AccountSearchWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: 'fasp', retry: 0
+
+  def perform(query)
+    return unless Mastodon::Feature.fasp_enabled?
+
+    async_refresh = AsyncRefresh.new("fasp:account_search:#{Digest::MD5.base64digest(query)}")
+
+    account_search_providers = Fasp::Provider.with_capability('account_search')
+    return if account_search_providers.none?
```

Comment:
This is the logic I was thinking of for the controller

Reply:
See my comment there.